lib/service/functions.js

```diff
@@ -40,18 +40,20 @@ function process(task, modules, callback) {
     return callback(task);
   }
 
-  const context = {};
-  const currentContext = task.hookType === 'post' ? task.response : task.request;
+  const { method, username, userId, entityId, loginOptions } = task.request;
+  const context = { method, username, userId, entityId, loginOptions };
+
+  for (const key in context) {
+    if (context[key] == undefined) delete context[key];
+  }
 
-  context.method = task.request.method;
+  const currentContext = task.hookType === 'post' ? task.response : task.request;
   context.headers = currentContext.headers;
-  context.username = task.request.username;
-  context.userId = task.request.userId;
   context.objectName = task.request.objectName || task.request.collectionName;
   context.hookType = task.hookType;
 
-  if (task.request.entityId) {
-    context.entityId = task.request.entityId;
+  if (task.hookType === 'post' && task.response.status > 399) {
+    context.status = task.response.status;
```

Comment:
It seems possible to combine the two checks of `task.hookType === 'post'` into one. Would be simpler IMO.

Reply:
Done